testing/architecture-test/src/changes/archunit-store/public-api-mutable-properties.txt

```diff
@@ -67,6 +67,7 @@ Method <org.gradle.api.artifacts.MinimalExternalModuleDependency.getModule()> do
 Method <org.gradle.api.artifacts.MinimalExternalModuleDependency.getVersionConstraint()> does not have raw return type (org.gradle.api.artifacts.VersionConstraint) assignable to any of [Provider] in (MinimalExternalModuleDependency.java:0)
 Method <org.gradle.api.artifacts.ModuleDependency.getArtifacts()> does not have raw return type (java.util.Set) assignable to any of [Provider] in (ModuleDependency.java:0)
 Method <org.gradle.api.artifacts.ModuleDependency.getAttributes()> does not have raw return type (org.gradle.api.attributes.AttributeContainer) assignable to any of [Provider] in (ModuleDependency.java:0)
+Method <org.gradle.api.artifacts.ModuleDependency.getCapabilitySelectors()> does not have raw return type (java.util.Set) assignable to any of [Provider] in (ModuleDependency.java:0)
```

Comment:
We made the explicit decision to not make this a provider. 

We do not expect users to be "wiring off" of dependency values. Dependencies should the the "leaf" of provider chains, and as such we provide a lazy way to _set_ capability selectors, but no lazy way to get them.

In any case, the provider API migration effort has made an explicit decision to _not_ migrate any properties of `Dependency` to providers, as as such we will follow that decision to not introduce this as a provider.